lib/adapter.js

```diff
@@ -7263,7 +7264,7 @@ function Adapter(options) {
             }
 
             // if pattern known, remove it from alias patterns to not subscribe to further matching aliases
-            for (const i in this.aliasPatterns) {
+            for (let i = 0; i < this.aliasPatterns.length; i++) {
```

Comment:
in `for (const i in array)`, the type of `i` is string. Alternatively, this could be a one-liner:
```
this.aliasPatterns = this.aliasPatterns.filter(pattern => pattern !== aliasPattern)
```